### PR TITLE
 Redundant null check on object #11223

### LINKED ISSUE #11223: Redundant null check on object
BUG

**What is the current behavior?**
There's a redundant null check on variable at [this line](https://github.com/Automattic/mongoose/blob/master/lib/document.js#L76), after checking type of variable as 'object'

**What is the expected behavior?**
It should perform the null check first, and if it passes null check then perform type check.

**What are the versions of Node.js, Mongoose and MongoDB you are using? Note that "latest" is not a version.**
Nodejs: 14.18.2
Mongoose: 6.1.6

### PR DESCRIPTION
**Pull request by dadwalakshay**
**Summary**
Fixed redundant null check on `skipId` variable at [this line](https://github.com/Automattic/mongoose/blob/master/lib/document.js#L76), by performing null check on mentioned variable before performing type check.

Resolves #11223

### CODE REVIEW

**Comment by Uzlopak** [comment]
First of all, this is not a redundant null check. So the title is already confusing. But this was already said in the linked issue.

And second of all, if this is for some "performance" improvement, than it makes no sense anyway. In most cases, skipId will not be null. The only reason the non null check is there is because we have to check if skipId is an Options-Object and if so switch the paramers for backwards compatibility reasons(?). But because null is also of type Object, we have to filter out false-positives like the null-Object. Your change assumes, that null is the most provided value for skipId, which is just an unproven assumption. It can not even increase the performance, even if this should be the "bottleneck" as we will probably fail more often in the typeof check than in the null-check. So your PR is making the situation worse, as it would decrease the performance.  

So in conclusion, this PR should be rejected.

**Comment by vkarpov15** [comment]
I'll close this PR. The null check isn't redundant, and the performance impact either way is negligible, so not worth worrying about any further.
